omicron_server/api_server.py

```diff
@@ -130,7 +154,14 @@ def revoke_token():
                 username=username_to_delete
             ).first()
             if user is None:
-                abort(404)
+                response = jsonify(
+                    {'error': 'unable to find user with username %s' %
+                              username_to_delete
+                     }
+                )
+                response.status_code = 404
```

Comment:
If the user is an administrator, they should be able to delete any user's token. This was solved here by passing in the username to delete as a query parameter. The corresponding request would look like

```/token?username=user```

It would make a lot more sense to move this to a separate endpoint, called ```/token/<token>```. a ```DELETE``` request to this endpoint would then check that the user that just authenticated is either an admin or the token's owner, and let the user delete it there. The 404 seems a more natural error to return in that context, than in the case of a bad query parameter. This endpoint would also be a bit more powerful